crates/matrix-sdk-common/src/tracing_timer.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::time::Instant;
+
+#[derive(Debug)]
+/// A named RAII that will show on Drop how long its covered section took to
+/// execute.
+pub struct TracingTimer {
+    id: String,
+    start: Instant,
+}
+
+impl Drop for TracingTimer {
+    fn drop(&mut self) {
+        tracing::debug!("{} finished in {}ms", self.id, self.start.elapsed().as_millis());
```

Comment:
Calling the macro in `Drop` means that all these logs will have a target of `matrix_sdk_common::tracing_timer`. Is that your intention? I think it might be possible to change this by having a *macro* for constructing the timer (could then also take care of declaring the guard binding) that constructs the event but doesn't emit it yet (pretty sure this is possible, not sure how conveniently).

Reply:
Could make something work by using a mix of what the tracing macros do, and what we already do in the FFI layer.